dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -130,10 +130,7 @@ internal sealed class KernelFunctionFromPrompt : KernelFunction
 
         var textContent = await textGeneration.GetTextContentAsync(renderedPrompt, arguments.ExecutionSettings, kernel, cancellationToken).ConfigureAwait(false);
 
-        IDictionary<string, object?> metadata = textContent.Metadata ?? new Dictionary<string, object?>();
-        metadata.Add(KernelEventArgsExtensions.RenderedPromptMetadataKey, renderedPrompt);
-
-        return new FunctionResult(this, textContent.Text, kernel.Culture, new Dictionary<string, object?>(metadata));
+        return new FunctionResult(this, textContent.Text, kernel.Culture, new Dictionary<string, object?>(textContent.Metadata ?? new Dictionary<string, object?>()));
```

Comment:
```suggestion
        return new FunctionResult(this, textContent.Text, kernel.Culture, textContent.Metadata);
```

Reply:
Actually this was good, I just looked and fix a potential reference leak between different choices and chunks. 

Knowing that each choice and chunk now will have its own dictionary, we can pass the reference to the FunctionResult.